### ISSUE TITLE
Improve error reporting: Missing "else" branch for if expression with elif

### ISSUE THREAD

**Issue by isaacabraham** [dotnet/fsharp#6873]
Related to #1104. Indeed in that issue there was actually a comment:

> This should also apply to if / elif / elif with a missing "else".

## What
Exactly the same old error message shows as soon as you add an `elif` branch e.g.

```fsharp
let x = 10
let y =
   if x > 10 then "test"
   elif x > 2 then "blah"
```

gives the error:

```
error FS0001: This expression was expected to have type
    'unit'    
but here has type
    'string'   
```

## Why
Same as #1104.

## How
Same as #1104.

**Comment by cartermp** [comment]
Thanks for filing 🙂

**Comment by isaacabraham** [comment]
@forki would you be interested in working with me on this one? Maybe a beifahrer screenshare or something to show me a bit more how I can do this?

**Comment by KathleenDollard** [comment]
Do you _always_ need the else? For example, you don't need it in a list comprehension do you? 

What do we need to discuss prior to moving this out of "needs discussion"?

**Comment by isaacabraham** [comment]
@KathleenDollard that's true re: list comprehension, although that's a somewhat specific case that (I believe) is really only because F# now supports implicit `yield`. In reality, such code would really be as follows:

Newer F#

```fsharp
let y = 1
let x = [
    if y = 2 then "a" // branch looks like it returns a string
    elif y = 3 then "b" // branch looks like it returns a string
]
```

is in reality this:

```fsharp
let y = 1
let x = [
    if y = 2 then yield "a" // branch returns unit
    elif y = 3 then yield "b" // branch returns unit
]
```

Since the expression is actually not returning anything you can omit the `else` branch. So, yes, I think that this should be approved as "up for grabs" :_)
